mcs/class/Mono.Data.Sqlite/Mono.Data.Sqlite_2.0/UnsafeNativeMethods.cs

```diff
@@ -16,6 +16,29 @@ namespace Mono.Data.Sqlite
 #endif
   internal static class UnsafeNativeMethods
   {
+    internal static readonly bool use_sqlite3_close_v2 = false;
+    internal static readonly bool use_sqlite3_open_v2 = false;
+    internal static readonly bool use_sqlite3_create_function_v2 = false;
+    static UnsafeNativeMethods()
+    {
+      // calculate the version number parts
+      // https://www.sqlite.org/c3ref/c_source_id.html
+      // (<major> * 1000000) + (<minor> * 1000) + (<release>)
+      int version = sqlite3_libversion_number();
+      int release = version % 1000;
+      int minor = (version / 1000) % 1000;
+      int major = version / 1000000;
+
+      // set the various versions
+      // https://sqlite.org/changes.html
+      bool v3_5_0 = major >= 3 && minor >= 5 && release >= 0;
+      bool v3_7_14 = major >= 3 && minor >= 7 && release >= 14;
+      bool v3_7_3 = major >= 3 && minor >= 7 && release >= 3;

```

Comment:
the bools will be false for e.g. version `3.8.0` or `4.0.0`, that looks wrong to me.

Consider using the Version class from the framework:

```
var version = new Version (major, minor, release);
use_sqlite3_open_v2 = version >= new Version (3, 5, 0);
...
```